TombEngine/Scripting/Internal/TEN/View/DisplayItem/ScriptDisplayItem.cpp

```diff
@@ -0,0 +1,671 @@
+	void ScriptDisplayItem::SetItemMeshRotation(int meshIndex, Rotation rotation, TypeOrNil<bool> disableInterpolation)
+	{
+		if (_itemName.empty())
+			return;
+
+		bool convertedBool = ValueOr<bool>(disableInterpolation, false);
+
+		auto* item = g_DrawItems.GetItemByName(_itemName);
+		
+		if (item)
+		item->SetItemMeshRotation(meshIndex, rotation.ToEulerAngles(), convertedBool);
+	}
+
+	/// Set the DisplayItems's visibility.
+	// @bool visible true if the item should become visible, false if it should become invisible.
+	// @function DisplayItem:SetVisible
+	// @usage
+	// local item = TEN.View.DisplayItem.GetItemByName("item1")
+	// item:SetVisible(true)
+	void ScriptDisplayItem::SetItemVisibility(bool visible)
+	{
+		if (_itemName.empty())
+			return;
+
+		auto* item = g_DrawItems.GetItemByName(_itemName);
+
+		if (item)
+			item->SetItemVisibility(visible);
+	}
+
+	/// Set frame number from an animation.
+	// This will set the specified animation to the given frame.
+	// The number of frames in an animation can be seen under the heading "End frame" in
+	// the WadTool animation editor.
+	// @function DisplayItem:SetFrame
+	// @tparam int animIndex The index of the desired animation.
+	// @tparam int frame The new frame number.
+	void ScriptDisplayItem::SetItemFrame(int animIndex, int frame)
+	{
+		if (_itemName.empty())
+			return;
+
+		auto* item = g_DrawItems.GetItemByName(_itemName);
+
+		if (item)
+		{
+			auto endFrame = GetEndFrame();
+
+			item->SetItemAnimation(animIndex);
+			if (frame <= endFrame)
+				item->SetItemFrame(frame);
+			else
+				item->SetItemFrame(endFrame);
+		}
+	}
+
+	/// Retrieve the object ID from a DisplayItem.
+	// @function DisplayItem:GetObjectID
+	// @treturn Objects.ObjID A number representing the object ID of the DisplayItem.
+	// @usage
+	// local item = TEN.View.DisplayItem.GetItemByName("item1")
+	// local objectID = item:GetObjectID()
+	GAME_OBJECT_ID ScriptDisplayItem::GetItemObjectID() const
+	{
+		auto* item = g_DrawItems.GetItemByName(_itemName);
+		
+		if (item)
+		return item->GetItemObjectID();
+
+		return ID_NO_OBJECT;
+	}
+	/// Get the DisplayItem's position.
+	// @function DisplayItem:GetPosition
+	// @treturn Vec3 DisplayItem's position.
+	// @usage
+	// local item = TEN.View.DisplayItem.GetItemByName("item1")
+	// local objectPosition = item:GetPosition()
+	Vec3 ScriptDisplayItem::GetItemPosition() const
+	{
+		if (_itemName.empty())
+			return Vec3(0, 0, 0);
+
+		auto* item = g_DrawItems.GetItemByName(_itemName);
+		if (!item)
+			return Vec3(0, 0, 0);
+
+		return Vec3(item->GetItemPosition());
+	}
+
+	/// Get the DisplayItem's rotation.
+	// @function DisplayItem:GetRotation
+	// @treturn Rotation DisplayItem's rotation.
+	// @usage
+	// local item = TEN.View.DisplayItem.GetItemByName("item1")
+	// local objectRotation = item:GetRotation()
+	Rotation ScriptDisplayItem::GetItemRotation() const
+	{
+		if (_itemName.empty())
+			return Rotation(0, 0, 0);
```

Comment:
Can you just use `Rotation()` here? Default constructor should initialize it with zeros.